tikv-client/src/main/java/com/pingcap/tikv/types/EnumType.java

```diff
@@ -92,9 +92,19 @@ private Integer parseEnumValue(Integer number) throws ConvertOverflowException {
   /** {@inheritDoc} */
   @Override
   protected Object decodeNotNull(int flag, CodecDataInput cdi) {
-    if (flag != Codec.UVARINT_FLAG)
-      throw new TypeException("Invalid EnumType(IntegerType) flag: " + flag);
-    int idx = (int) IntegerCodec.readUVarLong(cdi) - 1;
+    int idx;
+    switch (flag) {
```

Comment:
The comments are quite right. When we encode key, we need to make key comparable.  Do we have any need to decode a key? If so, it can be added as part of comments.

Reply:
If a column with enum type is declared as pk or unique index, then we need to decode a enum typed key.